corehq/apps/app_manager/templates/app_manager/partials/settings/app_settings.html

```diff
@@ -221,6 +221,9 @@ <h6>Example:</h6>
       <li><a href="#add-ons" data-toggle="tab">{% trans "Add-ons" %}</a></li>
     {% endif %}
     <li><a href="#commcare-settings" data-toggle="tab">{% trans "Advanced Settings" %}</a></li>
+    {% if not is_linked_app and not is_remote_app and request|toggle_enabled:'CUSTOM_ASSERTIONS' %}
+      <li><a href="#custom-assertions-tab" data-toggle="tab">{% trans "Custom Assertions" %}</a></li>
+    {% endif %}
```

Comment:
I ended up adding this as a whole new tab on the app settings page (see the screenshot in the PR description).  I really didn't want to do that, but it was the only approach I was able to get working.  The advanced settings page is controlled by a knockout model that generates the options dynamically from yaml files.  Sticking this in there gives a warning about multiple bindings being applied to the same element.  I think the answer there is to make this a knockout component that can be initialized from multiple places, but I wasn't able to get that working either.

Reply:
I haven't reviewed yet but wanted to check if you explored the `CUSTOM_PROPERTIES` feature flag? At least superficially, it looks similar: it adds a key value mapping widget to the advanced settings tab.

Reply:
I have - I initially tried adding it the same place, and I can add HTML in the same way, but the JS for that widget is controlled by the same knockout model as the generated-from-yaml stuff, whereas I'm trying to reuse the JS from elsewhere.